src/ProjectManager.h

```diff
@@ -125,6 +127,8 @@ class AUDACITY_DLL_API ProjectManager final
 
    // non-static data members
    AudacityProject &mProject;
+   ProjectTempoListener mTempoListener;
+   ProjectTempoPublisher mTempoPublisher;
```

Comment:
Again I don’t see why the publisher must be a member of ProjectManager.

I would just make it an independent attached object on the Project’s ClientData::Site.

And then it really needs only a .cpp file, no .h file, because it does all its magic by registering its factory at startup.

Reply:
Looks like it works. Coming up.